src/background.ts

```diff
@@ -67,46 +67,53 @@ async function generateQuery(message: { context: string, url: string, dims: [num
   graphicUrl = message.sourceURL
   var graphicWidth:number;
   var graphicHeight:number;
-    return fetch(message.sourceURL).then(resp => {
-        if (resp.ok) {
-          return resp.blob();
-        } else {
-          throw resp;
-        }
-    }).then(async(blobFile) => {
-      graphicWidth = message.dims[0];
-      graphicHeight = message.dims[1];
-      if(graphicWidth> 1200 && graphicWidth > graphicHeight){
-        message.dims[0] = 1200;
-        message.dims[1] = Math.round(graphicHeight*1200/graphicWidth);
-        return fromBlob(blobFile, message.dims[0], 'auto', 'webp');
-      } else if(graphicHeight > 1200){
-        message.dims[0] = Math.round(graphicWidth*1200/graphicHeight);
-        message.dims[1] = 1200;
-        return fromBlob(blobFile, message.dims[0], 'auto', 'webp');
+  return fetch(message.sourceURL).then(resp => {
+      if (resp.ok) {
+        return resp.blob();
       } else {
-        return blobFile;
+        throw resp;
       }
-    }).then(blob => {
-        return new Promise<string>((resolve, reject) => {
-            const reader = new FileReader();
-            reader.onload = () => resolve(reader.result as string);
-            reader.onerror = () => reject(reader.error);
-            reader.readAsDataURL(blob);
-        });
-    }).then(image => {
-        return {
-            "request_uuid": uuidv4(),
-            "timestamp": Math.round(Date.now() / 1000),
-            "URL": message.url,
-            "graphic": image,
-            "dimensions": message.dims,
-            "context": message.context,
-            "language": "en",
-            "capabilities": [],
-            "renderers": renderers
-        } as IMAGERequest;
-    });
+  }).then(async(blob: Blob) => {
+    graphicWidth = message.dims[0];
+    graphicHeight = message.dims[1];
+    if(graphicWidth> 1200 && graphicWidth > graphicHeight){
+      message.dims[0] = 1200;
+      message.dims[1] = Math.round(graphicHeight*1200/graphicWidth);
+    } else if(graphicHeight > 1200){
+      message.dims[0] = Math.round(graphicWidth*1200/graphicHeight);
+      message.dims[1] = 1200;
+    }
+    const blobFile = new File([blob], "buffer.jpg", {type: blob.type});
+    console.debug(`originalFile size ${blobFile.size / 1024 / 1024} MB`);
+    const options = {
+      maxSizeMB: 2,
```

Comment:
As discussed, please change maxSize to 4MB

Reply:
changed it!